openedx/core/djangoapps/schedules/resolvers.py

```diff
@@ -432,86 +435,77 @@ def schedules_for_bin(self):
+        site -- Site object that filtered Schedules will be a part of
+        effective_datetime -- effective date to get active courses.
+        target_datetime -- datetime that the User's Schedule's schedule_date_field value should fall under
+        course_id -- course id
+        override_recipient_email -- string email address that should receive all emails instead of the normal
+                                    recipient. (default: None)
+
+    Static attributes:
+        log_prefix -- a string to indentify this queue in logs
+        experience_filter -- a queryset filter used to select only the users who should be getting this message as part
+                             of their experience. This defaults to users without a specified experience type and those
+                             in the "recurring nudges and upgrade reminder" experience.
     """
     async_send_task = attr.ib()
     site = attr.ib()
+    effective_datetime = attr.ib()
     target_datetime = attr.ib()
     course_id = attr.ib()
     override_recipient_email = attr.ib(default=None)
-
-    log_prefix = 'Next Section Course Update'
-    experience_filter = Q(experience__experience_type=ScheduleExperience.EXPERIENCES.course_updates)
-
-    def send(self):  # lint-amnesty, pylint: disable=arguments-differ
-        schedules = self.get_schedules()
-        for (user, language, context) in schedules:
-            msg = CourseUpdate().personalize(
-                Recipient(
-                    user.id,
-                    self.override_recipient_email or user.email,
-                ),
-                language,
-                context,
-            )
-            LOG.info(
-                'Sending email to user: {} for course-key: {}'.format(
-                    user.username,
-                    self.course_id
-                )
-            )
-            with function_trace('enqueue_send_task'):
-                self.async_send_task.apply_async((self.site.id, str(msg)), retry=False)
+    experience_filter = None
+    log_prefix = ''
 
     def get_schedules(self):
         """
-        Grabs possible schedules that could receive a Course Next Section Update and if a
-        next section highlight is applicable for the user, yields information needed to
-        send the next section highlight email.
+        Get applicable schedule objects for the effective dates and course.
         """
-        target_date = self.target_datetime.date()
         course_duration = get_expected_duration(self.course_id)
         schedules = Schedule.objects.select_related('enrollment').filter(
-            self.experience_filter,
             enrollment__is_active=True,
             enrollment__course_id=self.course_id,
             enrollment__user__is_active=True,
-            start_date__gte=target_date - course_duration,
-            start_date__lt=target_date,
+            start_date__gte=self.effective_datetime - course_duration,
```

Comment:
For the `CourseSectionDueDateReminder`, could we retrieve the longest relative date from the course sections instead of relying on the course duration? It's more accurate and less error-prone.

_Side note: I considered simplifying `get_next_section_highlights` by removing the user binding. This way, we would retrieve all subsections and their relative due dates only once. Then, we would use the `has_access` function to check if a specific user can access it. However, there is the [INDIVIDUAL_DUE_DATES](https://github.com/openedx/edx-platform/blob/9db024c435e2e9f487b45e1c496f8f281eec1979/lms/envs/common.py#L307-L312) feature, so we need to retrieve these sections for each user to support it._

Reply:
@Agrendalath Just to clarify, we will take the longest relative due date from the course subsections and use it as the base to calculate the course_duration.

Should we still take the max of min_duration (4 weeks) and the longest relative due date to calculate this to avoid setting the duration too low or in the case of no relative due dates being set?

Also, to avoid getting all subsection dates here, we can filter by `start_date__gte=self.effective_datetime - max_duration (18 weeks)` and then retrieve the subsections for each user in `_get_upcoming_due_dates`.

Reply:
@navinkarkera,
> Just to clarify, we will take the longest relative due date from the course subsections and use it as the base to calculate the course_duration.

Correct.

> Should we still take the max of min_duration (4 weeks) and the longest relative due date to calculate this to avoid setting the duration too low or in the case of no relative due dates being set?

Let's set the `min_duration` to 1 week. If I remember correctly, it's possible to set the relative date to this value.

> Also, to avoid getting all subsection dates here, we can filter by start_date__gte=self.effective_datetime - max_duration (18 weeks) and then retrieve the subsections for each user in _get_upcoming_due_dates.

I believe that the upper limit is no longer applicable (https://github.com/openedx/edx-platform/pull/32094).

Reply:
> I believe that the upper limit is no longer applicable (https://github.com/openedx/edx-platform/pull/32094).

@Agrendalath Hmm, then it will be a bit tricky to get a course duration to filter schedules for all users including ones with individual due dates. Any ideas?

Reply:
@navinkarkera, we can add a note (with a short explanation) that this functionality does not support individual due dates.

Reply:
@Agrendalath Done, please take a look at the latest commit.